packages/node-core/src/utils/graphql.ts

```diff
@@ -112,5 +114,70 @@ export function getColumnOption(
       }
     };
   }
+  if (field.jsonInterface) {
+    columnOption.get = function () {
+      const dataValue = this.getDataValue(field.name);
+
+      if (!dataValue || !field.jsonInterface) {
+        return field.isArray ? [] : null;
+      }
+      return field.isArray
+        ? dataValue.map((v: any) => processNestedJson(field, v, 'Get'))
+        : processNestedJson(field, dataValue, 'Get');
+    };
+    columnOption.set = function (val: unknown) {
+      if (val === undefined || isNull(val)) {
+        this.setDataValue(field.name, null);
+        return;
+      }
+      if (isArray(val)) {
+        const setValue = val.length === 0 ? [] : val.map((v) => processNestedJson(field, v, 'Set'));
+        this.setDataValue(field.name, setValue);
+      } else if (isObject(val)) {
+        this.setDataValue(field.name, processNestedJson(field, val, 'Set'));
+      } else {
+        throw new Error(`input for Json type is only support object or array, got type ${typeof val}`);
+      }
+    };
+  }
   return columnOption;
 }
+
+type OperationType = 'Get' | 'Set';
+
+/***
+ * Process nested json set/get, output is same as input value type
+ * @param field
+ * @param value
+ * @param operationType
+ */
+function processNestedJson(
+  field: GraphQLEntityField | GraphQLJsonFieldType,
+  value: any,
+  operationType: OperationType
+): any {
+  // bigIntFields and nestJsonFields from this level in the entity/json
+  const bigIntFields = field.jsonInterface?.fields.filter((f) => f.type === 'BigInt');
+  const nestJsonFields = field.jsonInterface?.fields.filter((f) => f.jsonInterface);
+
+  const processBigIntFields = (value: any) => {
+    if (bigIntFields && bigIntFields.length) {
```

Comment:
```suggestion
    if (bigIntFields) {
```

Reply:
bigIntFields can be empty []

Reply:
If its empty then the `for` will not iterate anything